### ISSUE TITLE
pass through LF compiler option

### ISSUE THREAD

**Issue by dcsan** [TypeStrong/atom-typescript#545]
for the tsc option:

```
--newLine NEWLINE                  Specifies the end of line sequence to be used when emitting files: 'CRLF' (dos) or 'LF' (unix).
```

I added this to my tsconfig.json:

```
"compilerOptions": {
    "newLine": "LF"
```

however i still see lots of this on each commit of generated files:

```
warning: CRLF will be replaced by LF in bot1/src/mongo/MongoWrap-spec.js.
The file will have its original line endings in your working directory.
```

do these compiler options have to be explicitly passed through to tsc?

I added `"foo": "bar"` within atom, and got no errors, but running `tsc` at the command line did give:

    error TS5023: Unknown compiler option 'foo'.

so I assume they are NOT being passed from atom and have to be added to atom-typescript somewhere...?

**Comment by basarat** [comment]
> they are NOT being passed from atom and have to be added to atom-typescript somewhere...

Possible bug. Relevant code : https://github.com/TypeStrong/atom-typescript/blob/9a7680146e6b6ece5edaba3aebeeb9baa7b8a8cd/lib/main/tsconfig/tsconfig.ts#L266 

up for grabs

**Comment by matheo** [comment]
Sadly I'm suddenly getting CRLF in my transpiled files and I don't know why.

I guess it started when I added some typings containing CRLF (borisyankov/DefinitelyTyped#5881) but now that I cleaned them, it keeps using CRLF.

I've tried setting the `compilerOptions.newLine` to "LF" with no luck either :(

**Comment by cubuspl42** [comment]
Any news? I don't wanna see that CRs :(

**Comment by pinoyyid** [comment]
seems that adding `lineFeed: LF` to tsconfig.js is being ignored by atom-typescript. I've confirmed the format of tsconfig.ts is correct by running `tsc` which correctly emits LF, but compiling via atom-typescript emits CRLF

**Comment by Namek** [comment]
bugs me, too.

**Comment by pdfernhout** [comment]
I'm being stung by this too. :-(  Atom TypeScript apparently ignores the newLine option in tsconfig while tsc works as expected. This bug creates confusion about whether output files have really changed. However, the code pointed to above by @basarat does seem to do something with newLine which is in typescriptEnumMap, so it may not be something as simple as just an option not being copied over?

**Comment by liukun** [comment]
A dirty fix for 8.4.0 to use 'LF' is adding following `this.getNewLine` to file `~/.atom/packages/atom-typescript/dist/main/lang/core/languageServiceHost2.js` around line 126

    function LanguageServiceHost(config) {
        this.getNewLine = function() {return '\n';};
        var _this = this;

Then compiled files will end line with '\n'.

**Comment by pdfernhout** [comment]
@liukun Your workaround suggestion is working for me. Thanks!
